### PR TITLE
WT-11658 Generate code change report in html format

### PR DESCRIPTION
**Pull request by jeremythorp**

### CODE REVIEW

File: test/evergreen/code_change_report/code_change_report.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            lines = hunk['lines']
+            report.append("<table>\n")
+            report.append("  <tr>\n")
+            report.append("    <th>Count</th>\n")
+            report.append("    <th>Branches</th>\n")
+            report.append("    <th>=+-</th>\n")
+            report.append("    <th></th>\n")
+            report.append("  </tr>\n")
+            for line in lines:
+                print("line: " + str(line))
```
Comment (ajmorton) [inline test/evergreen/code_change_report/code_change_report.py:58]: Is this a debug line, or intentionally left in?

File: test/evergreen/code_change_report/code_change_report.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                        non_zero_count = get_non_zero_count(branch_info)
+                        code_colour = get_html_colour(non_zero_count, num_branches)
+                        report.append("    <td>\n")
+                        report.append("      <details>\n")
+                        report.append("      <summary>\n")
+                        report.append("      {} of {}\n".format(non_zero_count, num_branches))
+                        meter_high = len(branch_info) * 0.999
+                        report.append(
+                            "      <meter value=\"{}\" optimum=\"{}\" max=\"{}\" high=\"{}\" low=\"{}\"></meter>".
+                            format(non_zero_count, num_branches, num_branches, meter_high, meter_high))
```
Comment (ajmorton) [inline test/evergreen/code_change_report/code_change_report.py:92]: I'm not familiar with meters, but should this be
```suggestion
                            format(non_zero_count, num_branches, num_branches, meter_high, meter_low))
```
?

**Review by ajmorton — APPROVED** [review]
LGTM 👍. No need for re-review.

I have a few notes on the HTML report, but these can be covered in a separate ticket:
- It'd be nice to show the git diff information more clearly. The strike-through works but it takes some effort to distinguish between unchanged lines and new lines on first glance.
- When expanding the branch info there's no way to know which branch is the true path and which is the false path. This may be an artifact of gcovr though.
- There's a lot of vertical space between each line, especially between deleted lines. It'd help to be a bit more compact

File: test/evergreen/code_change_report/code_change_report.py (hunk 1 above)
Comment (jeremythorp) [reply test/evergreen/code_change_report/code_change_report.py:58]: That was a debugging/development line, which is no longer needed. I'll delete it.

File: test/evergreen/code_change_report/code_change_report.py (hunk 2 above)
Comment (jeremythorp) [reply test/evergreen/code_change_report/code_change_report.py:92]: That's deliberate. I wanted anything less than the top value to make the scale red, and the way to do that, apparently, is to set the low value to the high value/maximum. I'll add a comment explaining that though, as that would seem valuable.

Comment (jeremythorp) [reply test/evergreen/code_change_report/code_change_report.py:92]: I'll also create a meter_low value and assign it the value of meter_high as I think that will make it clearer.

**Pull request merged**